sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/InfluxDBPublisher.java

```diff
@@ -40,34 +43,72 @@
 
   private InfluxDBPublisher() {}
 
+  public static void publishNexmarkResults(
+      final Collection<Map<String, Object>> results, final InfluxDBSettings settings) {
+    publishWithCheck(settings, () -> publishNexmark(results, settings));
+  }
+
   public static void publishWithSettings(
       final Collection<NamedTestResult> results, final InfluxDBSettings settings) {
+    publishWithCheck(settings, () -> publishCommon(results, settings));
+  }
+
+  private static void publishWithCheck(
+      final InfluxDBSettings settings, final PublishFunction publishFunction) {
     requireNonNull(settings, "InfluxDB settings must not be null");
     if (isNoneBlank(settings.measurement, settings.database)) {
       try {
-        publish(results, settings);
-      } catch (final Exception exception) {
+        publishFunction.publish();
+      } catch (Exception exception) {
         LOG.warn("Unable to publish metrics due to error: {}", exception.getMessage(), exception);
       }
     } else {
       LOG.warn("Missing property -- measurement/database. Metrics won't be published.");
     }
   }
 
-  private static void publish(
-      final Collection<NamedTestResult> results, final InfluxDBSettings settings) throws Exception {
+  private static void publishNexmark(
+      final Collection<Map<String, Object>> results, final InfluxDBSettings settings)
+      throws Exception {
 
-    final HttpClientBuilder builder = HttpClientBuilder.create();
+    final HttpClientBuilder builder = provideHttpBuilder(settings);
+    final HttpPost postRequest = providePOSTRequest(settings);
+    final StringBuilder metricBuilder = new StringBuilder();
+    results.forEach(
+        map ->
+            metricBuilder
+                .append(map.get("measurement"))
```

Comment:
I'm wondering if it would make sense to extract the appends of keys and values to a method, but I can't find a nice and clean way of doing it. @kamilwu do you have any thoughts about it?

Reply:
How about adding a method `getKV` that would return a String in this format:" <key>=<value>", e.g. "runner=DataflowRunner"? This would reduce the number of appends.

Reply:
@pawelpasterz

Reply:
Yeah, makes sense, thanks!